backend/src/routes/users.ts

```diff
@@ -0,0 +1,47 @@
+import express, { NextFunction, Request, Response } from "express";
+import { decodeAuthToken } from "src/services/auth";
+import { AuthError } from "src/errors/auth";
+import { ServiceError } from "src/errors/service";
+// import mongoose from "mongoose";
+import { User } from "src/models/users";
+
+const router = express.Router();
+
+router.get("/api/whoami/:jwttoken", async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Instead of getting the JWT token from the request path itself, can we put the token in a request header? So a header like `Authorization: Bearer ey<rest of token>...`? This way, we can re-use the code for extracting the JWT token from the request header and decoding it to find the user, and make a custom middleware we can use on other routes.